nemo/collections/nlp/data/language_modeling/megatron/lm_adapted_t5_dataset.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from nemo.collections.nlp.data.language_modeling.megatron.gpt_dataset import GPTDataset
+
+
+class T5LMAdaptedDataset(GPTDataset):
+    """
+    Dataset for unlearning span corruption (https://arxiv.org/abs/2104.08691) in T5 models.
+    Corresponds to the prefix-LM objective in the T5 paper (Table 3 in https://arxiv.org/abs/1910.10683).
+    """
+
+    def __init__(
+        self, cfg, trainer, tokenizer, name, data_prefix, documents, indexed_dataset, num_samples, seed, **kwargs
+    ):
+        self.seq_length_encoder = cfg.data.seq_length
+        self.seq_length_decoder = cfg.data.seq_length_dec
+        self.tokenizer = tokenizer
+        super().__init__(
+            cfg,
+            trainer,
+            name,
+            data_prefix,
+            documents,
+            indexed_dataset,
+            num_samples,
+            self.seq_length_encoder + self.seq_length_decoder,
+            seed,
+        )
+
+    def __getitem__(self, idx):
+        text = super().__getitem__(idx)
+        text = text['text']
+
+        # Split text sequence into encoder and decoder inputs
+        tokens_enc = text[: self.seq_length_encoder]
+
+        # NOTE: Add bos only and not eos because the model will always generate till max seq length.
```

Comment:
Does it generate padding instead of `<eos>`? Will the extra generated tokens be masked (to prevent bias of gradients)?

Reply:
Any dataset that subclasses GPTDataset (like this one) and does not modifying it's `__getitem__` will not have any pad tokens.